aws/resource_aws_media_store_container.go

```diff
@@ -0,0 +1,149 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/mediastore"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsMediaStoreContainer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsMediaStoreContainerCreate,
+		Read:   resourceAwsMediaStoreContainerRead,
+		Update: resourceAwsMediaStoreContainerUpdate,
+		Delete: resourceAwsMediaStoreContainerDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if !regexp.MustCompile("^\\w+$").MatchString(value) {
+						errors = append(errors, fmt.Errorf("%q must match \\w+", k))
+					}
+					return
+				},
+			},
+			"policy": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ValidateFunc:     validateIAMPolicyJson,
+				DiffSuppressFunc: suppressEquivalentAwsPolicyDiffs,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsMediaStoreContainerCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).mediastoreconn
+
+	input := &mediastore.CreateContainerInput{
+		ContainerName: aws.String(d.Get("name").(string)),
+	}
+
+	_, err := conn.CreateContainer(input)
+	if err != nil {
+		return err
+	}
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{mediastore.ContainerStatusCreating},
+		Target:     []string{mediastore.ContainerStatusActive},
+		Refresh:    mediaStoreContainerRefreshStatusFunc(conn, d.Get("name").(string)),
+		Timeout:    10 * time.Minute,
+		Delay:      10 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("[WARN] Error waiting for MediaStore Container status to be \"ACTIVE\": %s", err)
+	}
+
+	d.SetId(d.Get("name").(string))
+	return resourceAwsMediaStoreContainerUpdate(d, meta)
+}
+
+func resourceAwsMediaStoreContainerRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).mediastoreconn
+
+	input := &mediastore.DescribeContainerInput{
+		ContainerName: aws.String(d.Id()),
+	}
+	resp, err := conn.DescribeContainer(input)
+	if err != nil {
+		return err
+	}
+	d.Set("arn", resp.Container.ARN)
+	d.Set("endpoint", resp.Container.Endpoint)
+	return nil
+}
+
+func resourceAwsMediaStoreContainerUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).mediastoreconn
+
+	if !d.HasChange("policy") {
+		return resourceAwsMediaStoreContainerRead(d, meta)
+	}
```

Comment:
I believe the above 3 lines will never be executed since all fields are `ForceNew` (so the only way the function ever gets executed is when `policy` has changed), unless I'm mistaken?

Reply:
I called `resourceAwsMediaStoreContainerUpdate` from the end of `resourceAwsMediaStoreContainerCreate` so I needed the above 3 lines.
Separating policy as `aws_media_store_container_policy` will solve this complication.

Reply:
Ah, I see - that's confusing. It actually makes me reconsider what I said in my other comment - I think we should remove all the policy related logic from this PR and implement it as a separate resource in another PR.

Do you agree?

Reply:
Yes, I agree👍 I'll remove